src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceMetadataBehavior.cs

```diff
@@ -139,11 +139,20 @@ private void SetExtensionProperties(ServiceDescription description, ServiceHostB
             mex.HttpGetUrl = host.GetVia(Uri.UriSchemeHttp, _httpGetUrl ?? new Uri(string.Empty, UriKind.Relative));
             mex.HttpsGetUrl = host.GetVia(Uri.UriSchemeHttps, _httpsGetUrl ?? new Uri(string.Empty, UriKind.Relative));
 
-            UseRequestHeadersForMetadataAddressBehavior dynamicUpdateBehavior = description.Behaviors.Find<UseRequestHeadersForMetadataAddressBehavior>();
-            if (dynamicUpdateBehavior != null)
+            UseRequestHeadersForMetadataAddressBehavior useRequestHeadersForMetadataAddressBehvaior = description.Behaviors.Find<UseRequestHeadersForMetadataAddressBehavior>();
+            if (useRequestHeadersForMetadataAddressBehvaior != null)
             {
```

Comment:
I think we could simplify this by having UseRequestHeadersForMetadataAddressBehavior provide an implementation of IEndpointAddressForMetadataProvider. We could get rid of UpdateAddressDynamically and have checks for that check mex.CustomEndpointAddressProvider != null instead. Of have UpdateAddressDynamically be defined as `get => CustomEndpointAddressProvider != null;`.

Reply:
done